### PR TITLE
Add tag repository support to krel stage

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
We now include the logic to tag a repository on `krel stage`.

#### Which issue(s) this PR fixes:

Refers to #1673 
#### Special notes for your reviewer:
/hold
for manual testing
#### Does this PR introduce a user-facing change?

```release-note
Added repository tagging functionality to `krel stage`
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
#### Manual testing:

:heavy_check_mark: Stage with modified `parentBranch == master` and simulating a new RC, including a new release branch:
https://console.cloud.google.com/cloud-build/builds/a3150055-7311-417d-967f-2ea12cbfbe7a?project=kubernetes-release-test
```
Step #2: INFO Configuring git user and email               
Step #2: INFO Preparing version v1.20.0-rc.0               
Step #2: INFO Parent branch provided: master               
Step #2: INFO Version v1.20.0-rc.0 is the prime version    
Step #2: INFO Creating or checking out release branch release-1.20 
Step #2: INFO Verifying release-1.20 branch exists in the repo 
Step #2: INFO Branch already exist: false                  
Step #2: INFO Creating release branch release-1.20 from commit 4628c605aadb9b 
Step #2: INFO Current branch is "release-1.20"             
Step #2: INFO Creating empty release commit for tag v1.20.0-rc.0 
Step #2: INFO Tagging version "v1.20.0-rc.0"               
Step #2: INFO Preparing version v1.21.0-alpha.0            
Step #2: INFO Parent branch provided: master               
Step #2: INFO Version v1.21.0-alpha.0 it not the prime, checking out parent branch 
Step #2: INFO Current branch is "master"                   
Step #2: INFO Tagging version "v1.21.0-alpha.0"  
````

- :hourglass: Stage for new beta:
https://console.cloud.google.com/cloud-build/builds/e5dbd39d-2cb6-4afb-ba64-73f9cb570b56?project=kubernetes-release-test

**Comment by saschagrunert** [comment]
Rebased on top of the latest master to resolve the conflict.

File: pkg/git/git.go [hunk 1]
```diff
@@ -54,6 +54,8 @@ const (
 	DefaultBranch            = "master"
 
 	defaultGithubAuthRoot = "git@github.com:"
+	defaultGitUser        = "Anago GCB"
+	defaultGitEmail       = "nobody@k8s.io"
```
Comment (justaugustus) [inline pkg/git/git.go:58]: We may want to consider changing these in future (post-implementation)

**Review by justaugustus — APPROVED** [review]
/lgtm
/approve

File: pkg/git/git.go (hunk 1 above)
Comment (saschagrunert) [reply pkg/git/git.go:58]: I agree :+1:

**Pull request merged**